client/app/src/main/java/com/microsoft/research/karya/ui/registration/CreationCodeFragment.kt

```diff
@@ -1,33 +1,81 @@
 import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.findNavController
 import com.microsoft.research.karya.R
+import com.microsoft.research.karya.data.service.KaryaAPIService
 import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.SeparatorTextWatcher
-import kotlinx.android.synthetic.main.activity_creation_code.*
+import kotlinx.android.synthetic.main.fragment_creation_code.*
+import kotlinx.android.synthetic.main.fragment_creation_code.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [CreationCodeFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+
 
 private const val CREATION_CODE_LENGTH = 16
 
-class CreationCodeActivity : BaseActivity(useAssistant = true) {
+class CreationCodeFragment : Fragment() {
 
     /** Compute creation code text box length based on the creation code length */
     private val creationCodeEtMax = CREATION_CODE_LENGTH + (CREATION_CODE_LENGTH - 1) / 4
 
     /** Android strings */
-    private var accessCodePromptMessage: String = ""
-    private var invalidCreationCodeMessage: String = ""
-    private var creationCodeAlreadyUsedMessage: String = ""
+
+    private lateinit var registrationActivity: RegistrationActivity
+    private lateinit var baseActivity: BaseActivity
+    private lateinit var karyaAPI: KaryaAPIService
+
+    protected val ioScope = CoroutineScope(Dispatchers.IO)
+    protected val uiScope = CoroutineScope(Dispatchers.Main)
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        setContentView(R.layout.activity_creation_code)
         super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        registrationActivity = activity as RegistrationActivity
+        baseActivity = activity as BaseActivity
+        karyaAPI = baseActivity.karyaAPI
+
+        /** Initialising Strings  **/
+        // TODO: Remove this implementation when we fetch strings from resource
+        // This Initialisation is only needed to be done once for the first fragment
+        baseActivity.initialise()
+
+        val fragmentView = inflater.inflate(R.layout.fragment_creation_code, container, false)
+
+        fragmentView.creationCodePromptTv.text = registrationActivity.accessCodePromptMessage
+
+        /** Inflating the layout for this fragment **/
+        return fragmentView
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        /** Initialise assistant audio **/
+        registrationActivity.current_assistant_audio = R.string.audio_access_code_prompt
```

Comment:
Maybe add this in TODO for now, we'll update the assistant later.

Reply:
Or we can have it in extensions? The code to actually run assistant is in BaseActivity

Reply:
Assistant method can be converted to an extension but then we'll have to create a new instance of `MediaPlayer` everytime we invoke the assistant so I'm not really sure about that.